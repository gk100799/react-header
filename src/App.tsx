import React, { Component } from 'react';
import { HashRouter as Router, Route, Switch } from 'react-router-dom';
import Header from './components/Header'
import { tabs } from './components/Constants'
import Cover from './components/Cover'

class App extends Component {
  render() {
    return (
      <Router >
        <div className="text-center root-class col-md-8 col-xs-12 col-sm-12 d-flex w-100 p-3 mx-auto flex-column">
          <Header />
          <Switch>
            {tabs && tabs.map(tab => (
              <Route
                key={tab.name}
                exact
                path={tab.path}
                component={() => <Cover {...tab.props} />}
              />
            ))}
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;
