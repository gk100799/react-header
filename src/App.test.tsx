import React from 'react';
import ReactDOM from 'react-dom'
import App from './App'
import { BrowserRouter as Router } from 'react-router-dom'
import { cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'

afterEach(cleanup);

const mockSomeConstantValueGetter = jest.fn();
jest.mock('./components/Constants.tsx', () => ({
    get tabs() {
        return mockSomeConstantValueGetter();
    },
}));

it("renders App.tsx without crashing", () => {
    const div = document.createElement("div")
    ReactDOM.render(<Router><App /></Router>, div)
})

it("matches snapshot", () => {
    const mockTabs = [
        {
            name: 'Home',
            path: '/',
            props: {
                coverHeading: 'This is Home page.',
                coverContent: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Totam magni illum molestiae reiciendis et quis libero porro neque suscipit beatae sunt facilis ullam corrupti temporibus voluptatibus, facere, quod corporis similique?',
                buttonTitle: 'Subscribe'
            }
        },
        {
            name: 'My Profile',
            path: '/profile',
            props: {
                coverHeading: 'This is My Profile page.',
                coverContent: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Accusamus mollitia, quae nihil a eius, quisquam quaerat deserunt obcaecati dolore earum ipsam ratione nobis culpa sequi! Obcaecati ab rerum alias nulla!',
                buttonTitle: 'Learn more'
            }
        }
    ]
    mockSomeConstantValueGetter.mockReturnValue(mockTabs);
    const tree = renderer.create(<Router><App /></Router>)
    expect(tree).toMatchSnapshot();
})