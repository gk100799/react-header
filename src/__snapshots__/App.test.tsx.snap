// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="text-center root-class col-md-8 col-xs-12 col-sm-12 d-flex w-100 p-3 mx-auto flex-column"
>
  <header
    className="masthead"
  >
    <div
      className="inner"
    >
      <h3
        className="masthead-brand"
        data-testid="logo"
      />
      <nav
        className="nav nav-masthead justify-content-center"
      >
        <a
          aria-current="page"
          className="nav-link active"
          data-testid="Home"
          href="#/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="nav-link"
          data-testid="My Profile"
          href="#/profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </nav>
    </div>
  </header>
  <div
    className="d-flex justify-content-center align-items-center h-100 w-100"
  >
    <main
      className="inner cover"
      role="main"
    >
      <h1
        className="cover-heading"
        data-testid="coverHeading"
      >
        This is Home page.
      </h1>
      <p
        className="lead"
        data-testid="coverContent"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Totam magni illum molestiae reiciendis et quis libero porro neque suscipit beatae sunt facilis ullam corrupti temporibus voluptatibus, facere, quod corporis similique?
      </p>
      <p
        className="lead"
      >
        <button
          className="btn btn-lg btn-secondary"
          data-testid="button"
        >
          Subscribe
        </button>
      </p>
    </main>
  </div>
</div>
`;
